lib/util/git/url.ts

```diff
@@ -29,12 +29,23 @@ export function getHttpUrl(url: string, token?: string): string {
 }
 
 export function getRemoteUrlWithToken(url: string, hostType?: string): string {
-  const hostRule = hostRules.find({ url, hostType });
+  const httpUrl = getHttpUrl(url)
+  const hostRule = hostRules.find({ url: httpUrl, hostType });
 
   if (hostRule?.token) {
     logger.debug(`Found hostRules token for url ${url}`);
-
-    return getHttpUrl(url, encodeURIComponent(hostRule.token));
+    const colonIndex = hostRule.token.indexOf(":")
```

Comment:
wrong, you should store as username and password instead. the this workaround isn't required.

Reply:
fortunately, github accepts `username: x-access-token, password: gh_123456789` as basic auth credentials for the API, so using username/password would technically work. however, I do think its worth looking into the issue with the token

Reply:
If username@host.name is valid, it's strange if simple-git or maybe git itself force a check for password.

But if that is as it is then you propose prepending `x-access-token:`? And is that for every type of token, both PAT and app tokens?

Reply:
I would like to check though - will a `hostRule` like 
```
{
    "matchHost": "github.com",
    "token": "ghs_abcd1234",
}
```
work on its own for release notes retrieval? we're populating these values using a script. I want to make sure that this is a valid token for release notes, and doesn't need to be of the form
```
{
    "matchHost": "github.com",
    "token": "x-access-token:ghs_abcd1234",
}
```

Reply:
In response to the above, from testing, it does appear to work fine. I would like to request re-review of this PR now :)